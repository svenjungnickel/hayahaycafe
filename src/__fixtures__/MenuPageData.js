export default {
    slug: 'slug',
    title: 'title',
    subtitle: 'subtitle',
    headerImage: {
        publicURL: 'publicURL',
    },
    images: [
        {
            src: {
                publicURL: 'publicURL',
            },
            alt: 'Menu page 01',
            title: 'Menu page 01',
        },
        {
            src: {
                publicURL: 'publicURL',
            },
            alt: 'Menu page 02',
            title: 'Menu page 02',
        },
        {
            src: {
                publicURL: 'publicURL',
            },
            alt: 'Menu page 03',
            title: 'Menu page 03',
        },
    ],
    meta: {
        title: 'meta title',
        description: 'meta description.',
        keywords: 'meta keywords',
    },
};
