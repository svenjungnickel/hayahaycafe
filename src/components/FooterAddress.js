import React from 'react';
import PropTypes from 'prop-types';
import { graphql, StaticQuery } from 'gatsby';

const AddressItem = ({ item }) => <>{item && <>{item}, </>}</>;

AddressItem.propTypes = {
    item: PropTypes.string.isRequired,
};

const FooterAddress = () => (
    <StaticQuery
        query={graphql`
            query {
                globalSettings: settingsYaml {
                    location {
                        address {
                            company
                            addressLine1
                            addressLine2
                            province
                            postalCode
                            country
                        }
                    }
                }
            }
        `}
        render={({ globalSettings }) => (
            <>
                {!!globalSettings && !!globalSettings.location && !!globalSettings.location.address && (
                    <p>
                        <AddressItem item={globalSettings.location.address.company} />
                        <AddressItem item={globalSettings.location.address.addressLine1} />
                        <AddressItem item={globalSettings.location.address.addressLine2} />
                        <AddressItem item={globalSettings.location.address.province} />
                        <AddressItem item={globalSettings.location.address.postalCode} />
                        {globalSettings.location.address.country && <>{globalSettings.location.address.country}</>}
                    </p>
                )}
            </>
        )}
    />
);

export default FooterAddress;
