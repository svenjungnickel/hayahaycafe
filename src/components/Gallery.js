import React, { useState } from 'react';
import PropTypes from 'prop-types';
import Carousel from 'nuka-carousel';
import FsLightbox from 'fslightbox-react';
import Image from './Image';
import GalleryStyle from '../styles/components/Gallery.module.scss';
import '../styles/libs/fslightbox-react.scss';

const PreviousButton = ({ previousSlide }) => (
    <button className={GalleryStyle.previousButton} aria-label="previous" onClick={previousSlide}>
        &lsaquo;
    </button>
);

PreviousButton.propTypes = {
    previousSlide: PropTypes.func.isRequired,
};

const NextButton = ({ nextSlide }) => (
    <button className={GalleryStyle.nextButton} aria-label="next" onClick={nextSlide}>
        &rsaquo;
    </button>
);

NextButton.propTypes = {
    previousSlide: PropTypes.func.isRequired,
};

const PagingDots = ({ slideCount, slidesToScroll, goToSlide, currentSlide }) => (
    <ul className={GalleryStyle.pagingDots__list}>
        {PagingDots.getIndexes(slideCount, slidesToScroll).map(index => {
            return (
                <li className={GalleryStyle.pagingDots__listItem} key={index}>
                    <button
                        className={GalleryStyle.pagingDots__button}
                        style={{ opacity: currentSlide === index ? 1 : 0.5 }}
                        onClick={goToSlide.bind(null, index)}
                    >
                        &bull;
                    </button>
                </li>
            );
        })}
    </ul>
);

PagingDots.getIndexes = (count, inc) => {
    const arr = [];
    for (let i = 0; i < count; i += inc) {
        arr.push(i);
    }
    return arr;
};

PagingDots.propTypes = {
    slideCount: PropTypes.number.isRequired,
    slidesToScroll: PropTypes.number.isRequired,
    goToSlide: PropTypes.func.isRequired,
    currentSlide: PropTypes.number.isRequired,
};

const Gallery = ({ images }) => {
    const [toggler, setToggler] = useState(false);
    const imageSources = images.map(image => {
        if (!!image.src && !!image.src.childImageSharp) {
            if (image.src.childImageSharp.fluid) {
                return image.src.childImageSharp.fluid.src;
            }

            return image.src.childImageSharp.fixed.src;
        }

        return image.src;
    });

    return (
        <>
            <Carousel
                dragging={true}
                swiping={true}
                wrapAround={true}
                heightMode="first"
                renderCenterLeftControls={props => <PreviousButton {...props} />}
                renderCenterRightControls={props => <NextButton {...props} />}
                renderBottomCenterControls={props => <PagingDots {...props} />}
            >
                {images.map((image, index) => (
                    <Image
                        key={index}
                        src={image.src}
                        alt={image.alt}
                        title={image.title}
                        onClick={() => setToggler(!toggler)}
                    />
                ))}
            </Carousel>
            <FsLightbox toggler={toggler} sources={imageSources} />
        </>
    );
};

Gallery.propTypes = {
    images: PropTypes.arrayOf(PropTypes.object).isRequired,
};

export default Gallery;
