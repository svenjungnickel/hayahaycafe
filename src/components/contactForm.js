import React, { useState, createRef } from 'react';
import { Alert, Button, Col, Form } from 'react-bootstrap';
import Recaptcha from 'react-google-recaptcha';
import Fade from 'react-reveal/Fade';
import Styles from '../styles/components/contactForm.module.scss';

const RECAPTCHA_KEY = process.env.SITE_RECAPTCHA_KEY;
const CONTACT_FORM_FADE_UP_DURATION = 400;
const SUCCESS_MESSAGE_FADE_IN_DURATION = 200;

function encode(data) {
    return Object.keys(data)
        .map(key => encodeURIComponent(key) + '=' + encodeURIComponent(data[key]))
        .join('&');
}

export default () => {
    const [validated, setValidated] = useState(false);
    const [state, setState] = useState({});
    const [recaptchaError, setRecaptchaError] = useState(false);
    const [success, setSuccess] = useState(false);
    const recaptchaRef = createRef();

    const handleChange = e => {
        setState({ ...state, [e.target.name]: e.target.value });
    };

    const handleRecaptcha = () => {
        setRecaptchaError(false);
    };

    const handleSubmit = event => {
        event.preventDefault();
        event.stopPropagation();
        setValidated(true);

        const recaptchaValue = recaptchaRef.current.getValue() || false;
        if (false === recaptchaValue) {
            setRecaptchaError(true);
            return;
        }

        if (event.currentTarget.checkValidity() === false) {
            return;
        }

        const form = event.target;
        fetch('/', {
            method: 'POST',
            headers: { 'Content-Type': 'application/x-www-form-urlencoded' },
            body: encode({
                'form-name': form.getAttribute('name'),
                'g-recaptcha-response': recaptchaValue,
                ...state,
            }),
        })
            .then(() => {
                console.log('success');
                setSuccess(true);
            })
            .catch(error => alert(error));
    };

    return (
        <div className={Styles.contactForm}>
            <Fade top collapse when={false === success} duration={CONTACT_FORM_FADE_UP_DURATION} unmountOnExit={true}>
                <Form
                    noValidate
                    validated={validated}
                    onSubmit={handleSubmit}
                    name="contact"
                    method="POST"
                    action="/thanks"
                    data-netlify="true"
                    data-netlify-recaptcha="true"
                >
                    <Form.Label>Name *</Form.Label>
                    <Form.Row>
                        <Col xs={12} lg={6}>
                            <Form.Group controlId="contactFirstName">
                                <Form.Control name="firstName" type="text" required onChange={handleChange} />
                                <Form.Text className="text-muted">First name</Form.Text>
                                <Form.Control.Feedback type="invalid">
                                    Please enter your first name.
                                </Form.Control.Feedback>
                            </Form.Group>
                        </Col>
                        <Col xs={12} lg={6}>
                            <Form.Group controlId="contactLastName">
                                <Form.Control name="lastName" type="text" required onChange={handleChange} />
                                <Form.Text className="text-muted">Last name</Form.Text>
                                <Form.Control.Feedback type="invalid">
                                    Please enter your last name.
                                </Form.Control.Feedback>
                            </Form.Group>
                        </Col>
                    </Form.Row>

                    <Form.Group controlId="contactEmail">
                        <Form.Label>Email *</Form.Label>
                        <Form.Control name="email" type="email" required onChange={handleChange} />
                        <Form.Control.Feedback type="invalid">
                            Please enter a valid email address.
                        </Form.Control.Feedback>
                    </Form.Group>

                    <Form.Group controlId="contactSubject">
                        <Form.Label>Subject *</Form.Label>
                        <Form.Control name="subject" type="text" required onChange={handleChange} />
                        <Form.Control.Feedback type="invalid">Please enter a subject.</Form.Control.Feedback>
                    </Form.Group>

                    <Form.Group controlId="contactMessage">
                        <Form.Label>Message *</Form.Label>
                        <textarea
                            name="message"
                            className="form-control"
                            id="contactMessage"
                            rows="5"
                            required
                            onChange={handleChange}
                        />
                        <Form.Control.Feedback type="invalid">Please enter your message.</Form.Control.Feedback>
                    </Form.Group>

                    <Form.Group controlId="contactRecaptcha">
                        <Recaptcha
                            id="contactRecaptcha"
                            ref={recaptchaRef}
                            sitekey={RECAPTCHA_KEY}
                            onChange={handleRecaptcha}
                            className={Styles.contactFormReCaptcha}
                        />
                        {true === recaptchaError && (
                            <div className="invalid-feedback" style={{ display: 'block' }}>
                                Please answer the captcha.
                            </div>
                        )}
                    </Form.Group>

                    <Button variant="primary" type="submit" className={Styles.contactFormSubmitButton}>
                        Submit
                    </Button>
                </Form>
            </Fade>
            <Fade
                collapse
                when={success}
                duration={SUCCESS_MESSAGE_FADE_IN_DURATION}
                delay={CONTACT_FORM_FADE_UP_DURATION}
            >
                <Alert variant="success" className={Styles.contactFormSuccess}>
                    <p>{`Thank you for your message. We're getting in touch with you soon.`}</p>
                </Alert>
            </Fade>
        </div>
    );
};
