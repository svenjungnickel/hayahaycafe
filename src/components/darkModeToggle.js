import React from 'react';
import useDarkMode from 'use-dark-mode';
import ReactSwitch from 'react-switch';

const Switch = props => <ReactSwitch {...props} />;

Switch.defaultProps = {
    checkedIcon: false,
    uncheckedIcon: false,
    height: 24,
    width: 48,
    handleDiameter: 24,
    offColor: `#000`,
    onColor: `#000`,
    boxShadow: `inset 0 0 0 1px #000`,
};

const uncheckedIcon = (
    <div
        style={{
            display: 'flex',
            justifyContent: 'center',
            alignItems: 'center',
            height: '100%',
            fontSize: 15,
            color: 'yellow',
            paddingLeft: 2,
        }}
    >
        ☀
    </div>
);

const checkedIcon = (
    <div
        style={{
            display: 'flex',
            justifyContent: 'center',
            alignItems: 'center',
            height: '100%',
            fontSize: 15,
            color: 'yellow',
            paddingRight: 2,
        }}
    >
        ☾
    </div>
);

const DarkModeToggle = props => {
    const darkMode = useDarkMode(false),
        darkModeValue = null !== darkMode.value ? darkMode.value : false;

    return (
        <Switch
            {...props}
            title="dark mode switch"
            aria-checked={darkModeValue}
            checked={darkModeValue}
            onChange={darkMode.toggle}
            uncheckedIcon={uncheckedIcon}
            checkedIcon={checkedIcon}
        />
    );
};

export default DarkModeToggle;
