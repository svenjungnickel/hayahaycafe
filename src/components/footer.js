import React from 'react';
import { Container, Nav } from 'react-bootstrap';
import SocialMediaIcons from './socialMediaIcons';
import FooterStyles from '../styles/components/footer.module.scss';

export default ({ siteMetadata }) => (
    <footer className={FooterStyles.footer}>
        <div className={FooterStyles.footer__pre}>
            <SocialMediaIcons items={siteMetadata.socialMedia} />
        </div>
        <div className={FooterStyles.footer__main}>
            <Container className="text-center">
                <Nav className="justify-content-center">
                    <Nav.Item>
                        <Nav.Link href="/contact">Contact</Nav.Link>
                    </Nav.Item>
                    <Nav.Item>
                        <Nav.Link href="/legal">Legal</Nav.Link>
                    </Nav.Item>
                    <Nav.Item>
                        <Nav.Link href="/data-privacy">Data Privacy</Nav.Link>
                    </Nav.Item>
                </Nav>

                <div className={FooterStyles.footer__address}>
                    <p>{siteMetadata.address}</p>
                </div>
            </Container>
        </div>
    </footer>
);
