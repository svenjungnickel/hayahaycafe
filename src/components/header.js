import React from 'react';
import Img from 'gatsby-image';
import PropTypes from 'prop-types';
import { Container } from 'react-bootstrap';
import IndexStyles from '../styles/components/header.module.scss';

const Header = ({ header, title, subtitle }) => (
    <header className={IndexStyles.header}>
        <Img fluid={header} alt="header" className={IndexStyles.header__image} />

        <div className={IndexStyles.header__content}>
            <Container>
                <div className={IndexStyles.header__contentInner}>
                    <h1 className="display-1">{title}</h1>
                    <p className="lead">{subtitle}</p>
                </div>
            </Container>
        </div>
    </header>
);

Header.propTypes = {
    header: PropTypes.object.isRequired,
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string.isRequired,
};

export default Header;
