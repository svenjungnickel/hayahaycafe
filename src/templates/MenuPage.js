import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { graphql } from 'gatsby';
import { Col, Container, Row } from 'react-bootstrap';
import { getSrc } from 'gatsby-plugin-image';
import Layout from '../components/Layout';
import Header from '../components/Header';
import Content, { HTMLContent } from '../components/Content';
import Separator from '../components/Separator';
import Image from '../components/Image';
import FsLightbox from 'fslightbox-react';
import { section } from '../styles/components/Section.module.scss';
import { menu, menu__image } from '../styles/pages/Menu.module.scss';

export const MenuPageTemplate = ({ title, subtitle, headerImage, content, contentComponent, images }) => {
    const PostContent = contentComponent || Content;

    const [lightBoxController, setLightBoxController] = useState({
        toggler: false,
        slideIndex: 0,
    });

    const imageSources = images.map((image) => {
        const imageSrc = getSrc(image.src);
        if (imageSrc) {
            return imageSrc;
        }

        return image.src;
    });

    const openLightBox = (index) => {
        if (window.matchMedia('(width < 992px)').matches === true) {
            return;
        }

        setLightBoxController({
            toggler: !lightBoxController.toggler,
            index: index,
        });
    };

    return (
        <>
            <Header headerImage={headerImage} title={title} subtitle={subtitle} />

            <section className={section}>
                <Container className={menu}>
                    <Row>
                        {!!content && (
                            <Col xs={12} data-cy="menuContent">
                                <PostContent content={content} />
                                <Separator />
                            </Col>
                        )}
                        {images && images.length > 0 && (
                            <>
                                {images.map((image, index) => (
                                    <Col md={6} lg={4} key={index} className={menu__image} data-cy="menuImage">
                                        <div onClick={() => openLightBox(index)}>
                                            <Image src={image.src} alt={image.alt} title={image.title} />
                                        </div>
                                    </Col>
                                ))}
                                <FsLightbox
                                    toggler={lightBoxController.toggler}
                                    sources={imageSources}
                                    sourceIndex={lightBoxController.index}
                                />
                            </>
                        )}
                    </Row>
                </Container>
            </section>
        </>
    );
};

MenuPageTemplate.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string.isRequired,
    headerImage: PropTypes.oneOfType([PropTypes.object, PropTypes.string]).isRequired,
    content: PropTypes.node.isRequired,
    contentComponent: PropTypes.func,
    images: PropTypes.arrayOf(PropTypes.object).isRequired,
};

const MenuPage = ({ data: { page } }) => (
    <Layout meta={page.frontmatter.meta} currentPage={page.frontmatter.slug}>
        <MenuPageTemplate {...page.frontmatter} content={page.html} contentComponent={HTMLContent} />
    </Layout>
);

MenuPage.propTypes = {
    data: PropTypes.shape({
        page: PropTypes.object.isRequired,
    }).isRequired,
};

export default MenuPage;

export const pageQuery = graphql`
    query MenuPage($id: String!) {
        page: markdownRemark(id: { eq: $id }) {
            html
            frontmatter {
                slug
                title
                subtitle
                headerImage {
                    childImageSharp {
                        gatsbyImageData(layout: FULL_WIDTH)
                    }
                    publicURL
                }
                images {
                    src {
                        childImageSharp {
                            gatsbyImageData(layout: CONSTRAINED, width: 1138, height: 1600)
                        }
                        publicURL
                    }
                    alt
                    title
                }
                meta {
                    title
                    description
                    keywords
                }
            }
        }
    }
`;
