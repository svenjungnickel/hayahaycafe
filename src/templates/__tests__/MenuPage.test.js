import React from 'react';
import renderer from 'react-test-renderer';
import { getSrc } from 'gatsby-plugin-image';
import MenuPage, { MenuPageTemplate } from '../MenuPage';
import { HTMLContent } from '../../components/Content';
import MenuPageData from '../../__fixtures__/MenuPageData';

jest.mock('../../components/Layout');
jest.mock('../../components/Header');
jest.mock('../../components/Separator');
jest.mock('../../components/Image');

describe('MenuPageTemplate', () => {
    it('renders without title throws prop type error', () => {
        const renderMenuPageTemplate = () => {
            const subtitle = 'subtitle';
            const headerImage = 'headerImage';
            const content = 'content';
            const images = [{}];

            const component = (
                <MenuPageTemplate subtitle={subtitle} headerImage={headerImage} content={content} images={images} />
            );
            renderer.create(component);
        };

        expect(renderMenuPageTemplate).toThrowError('Warning: Failed prop type');
    });

    it('renders without subtitle throws prop type error', () => {
        const renderMenuPageTemplate = () => {
            const title = 'title';
            const headerImage = 'headerImage';
            const content = 'content';
            const images = [{}];

            const component = (
                <MenuPageTemplate title={title} headerImage={headerImage} content={content} images={images} />
            );
            renderer.create(component);
        };

        expect(renderMenuPageTemplate).toThrowError('Warning: Failed prop type');
    });

    it('renders without headerImage throws prop type error', () => {
        const renderMenuPageTemplate = () => {
            const title = 'title';
            const subtitle = 'subtitle';
            const content = 'content';
            const images = [{}];

            const component = <MenuPageTemplate title={title} subtitle={subtitle} content={content} images={images} />;
            renderer.create(component);
        };

        expect(renderMenuPageTemplate).toThrowError('Warning: Failed prop type');
    });

    it('renders without content throws prop type error', () => {
        const renderMenuPageTemplate = () => {
            const title = 'title';
            const subtitle = 'subtitle';
            const headerImage = 'headerImage';
            const images = [{}];

            const component = (
                <MenuPageTemplate title={title} subtitle={subtitle} headerImage={headerImage} images={images} />
            );
            renderer.create(component);
        };

        expect(renderMenuPageTemplate).toThrowError('Warning: Failed prop type');
    });

    it('renders without images throws prop type error', () => {
        const renderMenuPageTemplate = () => {
            const title = 'title';
            const subtitle = 'subtitle';
            const headerImage = 'headerImage';
            const content = 'content';

            const component = (
                <MenuPageTemplate title={title} subtitle={subtitle} headerImage={headerImage} content={content} />
            );
            renderer.create(component);
        };

        expect(renderMenuPageTemplate).toThrowError('Warning: Failed prop type');
    });

    it('renders with invalid title throws prop type error', () => {
        const renderMenuPageTemplate = () => {
            const title = 123;
            const subtitle = 'subtitle';
            const headerImage = 'headerImage';
            const content = 'content';
            const images = [{}];

            const component = (
                <MenuPageTemplate
                    title={title}
                    subtitle={subtitle}
                    headerImage={headerImage}
                    content={content}
                    images={images}
                />
            );
            renderer.create(component);
        };

        expect(renderMenuPageTemplate).toThrowError('Warning: Failed prop type');
    });

    it('renders with invalid subtitle throws prop type error', () => {
        const renderMenuPageTemplate = () => {
            const title = 'title';
            const subtitle = 123;
            const headerImage = 'headerImage';
            const content = 'content';
            const images = [{}];

            const component = (
                <MenuPageTemplate
                    title={title}
                    subtitle={subtitle}
                    headerImage={headerImage}
                    content={content}
                    images={images}
                />
            );
            renderer.create(component);
        };

        expect(renderMenuPageTemplate).toThrowError('Warning: Failed prop type');
    });

    it('renders with invalid headerImage throws prop type error', () => {
        const renderMenuPageTemplate = () => {
            const title = 'title';
            const subtitle = 'subtitle';
            const headerImage = 123;
            const content = 'content';
            const images = [{}];

            const component = (
                <MenuPageTemplate
                    title={title}
                    subtitle={subtitle}
                    headerImage={headerImage}
                    content={content}
                    images={images}
                />
            );
            renderer.create(component);
        };

        expect(renderMenuPageTemplate).toThrowError('Warning: Failed prop type');
    });

    it('renders with invalid content throws prop type error', () => {
        const renderMenuPageTemplate = () => {
            const title = 'title';
            const subtitle = 'subtitle';
            const headerImage = 'headerImage';
            const content = {};
            const images = [{}];

            const component = (
                <MenuPageTemplate
                    title={title}
                    subtitle={subtitle}
                    headerImage={headerImage}
                    content={content}
                    images={images}
                />
            );
            renderer.create(component);
        };

        expect(renderMenuPageTemplate).toThrowError('Warning: Failed prop type');
    });

    it('renders with invalid contentComponent throws prop type error', () => {
        const renderMenuPageTemplate = () => {
            const title = 'title';
            const subtitle = 'subtitle';
            const headerImage = 'headerImage';
            const content = 'content';
            const contentComponent = false;
            const images = [{}];

            const component = (
                <MenuPageTemplate
                    title={title}
                    subtitle={subtitle}
                    headerImage={headerImage}
                    content={content}
                    contentComponent={contentComponent}
                    images={images}
                />
            );
            renderer.create(component);
        };

        expect(renderMenuPageTemplate).toThrowError('Warning: Failed prop type');
    });

    it('renders with invalid images throws prop type error', () => {
        const renderMenuPageTemplate = () => {
            const title = 'title';
            const subtitle = 'subtitle';
            const headerImage = 'headerImage';
            const content = 'content';
            const images = false;

            const component = (
                <MenuPageTemplate
                    title={title}
                    subtitle={subtitle}
                    headerImage={headerImage}
                    content={content}
                    images={images}
                />
            );
            renderer.create(component);
        };

        expect(renderMenuPageTemplate).toThrowError('Warning: Failed prop type');
    });

    it('renders content correctly', () => {
        const title = 'title';
        const subtitle = 'subtitle';
        const headerImage = 'headerImage';
        const content = 'content';
        const images = [{}];

        const component = (
            <MenuPageTemplate
                title={title}
                subtitle={subtitle}
                headerImage={headerImage}
                content={content}
                images={images}
            />
        );
        const tree = renderer.create(component).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('renders html content correctly', () => {
        const title = 'title';
        const subtitle = 'subtitle';
        const headerImage = 'headerImage';
        const content = <p>HTML content</p>;
        const images = [{}];

        const component = (
            <MenuPageTemplate
                title={title}
                subtitle={subtitle}
                headerImage={headerImage}
                content={content}
                contentComponent={HTMLContent}
                images={images}
            />
        );
        const tree = renderer.create(component).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('renders GatsbyImage', () => {
        const title = 'title';
        const subtitle = 'subtitle';
        const headerImage = 'headerImage';
        const content = 'content';
        const images = [{ src: { publicURL: 'publicURL' } }];

        getSrc.mockImplementation(() => true);

        const component = (
            <MenuPageTemplate
                title={title}
                subtitle={subtitle}
                headerImage={headerImage}
                content={content}
                images={images}
            />
        );
        const tree = renderer.create(component).toJSON();

        expect(tree).toMatchSnapshot();
    });
});

describe('MenuPage', () => {
    it('renders without data throws prop type error', () => {
        const renderStartPage = () => {
            const component = <MenuPage />;
            renderer.create(component);
        };

        expect(renderStartPage).toThrowError('Warning: Failed prop type');
    });

    it('renders with invalid data throws prop type error', () => {
        const renderStartPage = () => {
            const data = {};

            const component = <MenuPage data={data} />;
            renderer.create(component);
        };

        expect(renderStartPage).toThrowError('Warning: Failed prop type');
    });

    it('renders correctly', () => {
        const data = {
            page: {
                html: <p>HTML</p>,
                frontmatter: MenuPageData,
            },
        };

        const component = <MenuPage data={data} />;
        const tree = renderer.create(component).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
