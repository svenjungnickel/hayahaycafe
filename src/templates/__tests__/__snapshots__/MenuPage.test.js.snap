// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuPage renders correctly 1`] = `
<main>
  <header />
  <section
    className="section"
  >
    <div
      className="menu container"
    >
      <div
        className="row"
      >
        <div
          className="col-12"
          data-cy="menuContent"
        >
          <div
            dangerouslySetInnerHTML={
              {
                "__html": <p>
                  HTML
                </p>,
              }
            }
          />
          <hr />
        </div>
        <div
          className="menu__image col-lg-6 col-md-12"
          data-cy="menuImage"
        >
          <div
            onClick={[Function]}
          >
            <img
              alt="alt"
              src="src"
            />
          </div>
        </div>
        <div
          className="menu__image col-lg-6 col-md-12"
          data-cy="menuImage"
        >
          <div
            onClick={[Function]}
          >
            <img
              alt="alt"
              src="src"
            />
          </div>
        </div>
      </div>
    </div>
  </section>
</main>
`;

exports[`MenuPageTemplate renders GatsbyImage 1`] = `
[
  <header />,
  <section
    className="section"
  >
    <div
      className="menu container"
    >
      <div
        className="row"
      >
        <div
          className="col-12"
          data-cy="menuContent"
        >
          <div>
            content
          </div>
          <hr />
        </div>
        <div
          className="menu__image col-lg-6 col-md-12"
          data-cy="menuImage"
        >
          <div
            onClick={[Function]}
          >
            <img
              alt="alt"
              src="src"
            />
          </div>
        </div>
      </div>
    </div>
  </section>,
]
`;

exports[`MenuPageTemplate renders content correctly 1`] = `
[
  <header />,
  <section
    className="section"
  >
    <div
      className="menu container"
    >
      <div
        className="row"
      >
        <div
          className="col-12"
          data-cy="menuContent"
        >
          <div>
            content
          </div>
          <hr />
        </div>
        <div
          className="menu__image col-lg-6 col-md-12"
          data-cy="menuImage"
        >
          <div
            onClick={[Function]}
          >
            <img
              alt="alt"
              src="src"
            />
          </div>
        </div>
      </div>
    </div>
  </section>,
]
`;

exports[`MenuPageTemplate renders html content correctly 1`] = `
[
  <header />,
  <section
    className="section"
  >
    <div
      className="menu container"
    >
      <div
        className="row"
      >
        <div
          className="col-12"
          data-cy="menuContent"
        >
          <div
            dangerouslySetInnerHTML={
              {
                "__html": <p>
                  HTML content
                </p>,
              }
            }
          />
          <hr />
        </div>
        <div
          className="menu__image col-lg-6 col-md-12"
          data-cy="menuImage"
        >
          <div
            onClick={[Function]}
          >
            <img
              alt="alt"
              src="src"
            />
          </div>
        </div>
      </div>
    </div>
  </section>,
]
`;
